### ISSUE TITLE
Setup SQLSTATE Access denied

### ISSUE THREAD

**Issue by Hi-ImKyle** [AzuraCast/AzuraCast#5980]
### Installation Method

Docker Installation

### AzuraCast Release Channel

Rolling Release Channel

### Current AzuraCast Version

0.17.6

### What happened?

My attempt at installing AzuraCast through the [docker instructions ](https://docs.azuracast.com/en/getting-started/installation/docker#installing) gets to the end just for it to fail complaining about `Access Denied` causing HTTP 502 errors on the web page. This was the second attempt hence why the installer shows the config ports to be different than default. Before running the installer again I removed all AzuraCast docker containers/images to make sure it got everything fresh.

### Relevant log output

```shell
root@Squirtle /var/azuracast # curl -fsSL https://raw.githubusercontent.com/AzuraCast/AzuraCast/main/docker.sh > docker.sh
root@Squirtle /var/azuracast # chmod a+x docker.sh
root@Squirtle /var/azuracast # ./docker.sh install
Checking installation requirements for AzuraCast...
[PASS] Operating System: Linux
[PASS] Architecture: x86_64
[PASS] Command Present: curl
[PASS] Command Present: awk
[PASS] User Permissions
[PASS] Installation Directory
[PASS] All requirements met!
Docker is already installed! Continuing...
Docker Compose is already installed. Continuing...
Your current release channel is 'Stable'. Switch to 'Rolling Release' release channel? [y/N]
Pulling installer ... done

Creating azuracast_installer_installer_run ... done

 Select Language [English (Default)]:
  [en_US] English (Default)
  [cs_CZ] čeština
  [de_DE] Deutsch
  [es_ES] Español
  [fr_FR] Français
  [el_GR] ελληνικά
  [it_IT] Italiano
  [hu_HU] magyar
  [nl_NL] Nederlands
  [pl_PL] Polski
  [pt_PT] Português
  [pt_BR] Português do Brasil
  [ru_RU] Русский язык
  [sv_SE] Svenska
  [tr_TR] Türkçe
  [zh_CN] 簡化字
  [ko_KR] 한국어
 >

AzuraCast Installer
===================

 Welcome to AzuraCast! Complete the initial server setup by answering a few
 questions.

AzuraCast is currently configured to listen on the following ports:
 * HTTP Port: 10000
 * HTTPS Port: 10443
 * SFTP Port: 10022
 * Radio Ports: 10500,10505,10506,10510,10515,10516,10520,10525,10526,10530,10535,10536,10540,10545,10546

 Customize ports used for AzuraCast? (yes/no) [no]:
 > yes

 HTTP Port - The main port AzuraCast listens to for insecure HTTP connections. [10000]:
 > 4000

 HTTPS Port - The main port AzuraCast listens to for secure HTTPS connections. [10443]:
 > 4443

 SFTP Port - The port AzuraCast listens to for SFTP file management connections. [10022]:
 > 4022

 Minimum Port for Station Port Assignment [8000]:
 > 4500

 Maximum Port for Station Port Assignment [8499]:
 > 4550

 Enable Custom Code Plugins (yes/no) [no]:
 > yes

Writing configuration files...


 [OK] Server configuration complete!


Pulling web ... done
Creating network "azuracast_default" with the default driver
Creating azuracast_web_run ... done
** Running startup script '/etc/my_init.d/00_disable_mariadb.sh'...
** Startup script complete.
** Running startup script '/etc/my_init.d/00_setup_user.sh'...
usermod: no changes
Docker 'azuracast' User UID: 1000
Docker 'azuracast' User GID: 1000
** Startup script complete.
** Running startup script '/etc/my_init.d/01_self_signed_ssl.sh'...
Generating self-signed certificate...
~removed~
-----
** Startup script complete.
** Running startup script '/etc/my_init.d/02_prepare_cron.sh'...
** Startup script complete.
** Running startup script '/etc/my_init.d/04_mariadb_conf.sh'...
** Startup script complete.
** Running startup script '/etc/my_init.d/05_nginx_conf.sh'...
** Startup script complete.
** Running startup script '/etc/my_init.d/05_setup_db.sh'...
2022-12-31 01:06:39+00:00 [Note] [Entrypoint]: Initial DB setup...
2022-12-31 01:06:39+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
2022-12-31 01:06:39+00:00 [Note] [Entrypoint]: Initial DB setup...
2022-12-31 01:06:39+00:00 [Note] [Entrypoint]: MariaDB upgrade not required
** Startup script complete.
** Running startup script '/etc/my_init.d/07_sftpgo_conf.sh'...
** Startup script complete.
** Running startup script '/etc/my_init.d/10_clear_temp.sh'...
** Startup script complete.
** Running Supervisord...
** Supervisord started.
** Running uptime_wait...
Starting up AzuraCast services...
🖥️🎶－🎵－📻 Waiting...Timed out waiting for services to start.
** Services up and ready!
This script must be run as the "azuracast" user. Rerunning...
AzuraCast Setup
(Environment: production)
Loading composer repositories with package information
Info from https://repo.packagist.org: #StandWithUkraine
Updating dependencies
Lock file operations: 0 installs, 68 updates, 0 removals
  - Upgrading aws/aws-sdk-php (3.247.2 => 3.255.7)
  - Upgrading bacon/bacon-qr-code (2.0.7 => 2.0.8)
  - Upgrading codeception/codeception (5.0.5 => 5.0.6)
  - Upgrading codeception/stub (4.0.2 => 4.1.0)
  - Upgrading doctrine/annotations (1.13.3 => 1.14.2)
  - Upgrading doctrine/collections (1.8.0 => 2.1.2)
  - Upgrading doctrine/data-fixtures (1.5.3 => 1.6.1)
  - Upgrading doctrine/dbal (3.5.1 => 3.5.2)
  - Upgrading doctrine/instantiator (1.4.1 => 1.5.0)
  - Upgrading doctrine/lexer (1.2.3 => 2.1.0)
  - Upgrading doctrine/orm (2.13.4 => 2.14.0)
  - Upgrading doctrine/persistence (3.1.0 => 3.1.2)
  - Upgrading egulias/email-validator (3.2.1 => 3.2.4)
  - Upgrading laminas/laminas-code (4.7.1 => 4.8.0)
  - Upgrading league/flysystem (3.10.4 => 3.12.0)
  - Upgrading nesbot/carbon (2.63.0 => 2.64.0)
  - Upgrading pagerfanta/core (v3.6.2 => v3.7.0)
  - Upgrading pagerfanta/doctrine-collections-adapter (v3.6.2 => v3.7.0)
  - Upgrading pagerfanta/doctrine-orm-adapter (v3.6.2 => v3.7.0)
  - Upgrading php-ffmpeg/php-ffmpeg (v1.0.1 => v1.1.0)
  - Upgrading phpseclib/phpseclib (3.0.17 => 3.0.18)
  - Upgrading phpstan/phpdoc-parser (1.13.1 => 1.15.3)
  - Upgrading phpstan/phpstan (1.9.2 => 1.9.4)
  - Upgrading phpstan/phpstan-doctrine (1.3.23 => 1.3.28)
  - Upgrading phpunit/php-code-coverage (9.2.19 => 9.2.23)
  - Upgrading phpunit/phpunit (9.5.26 => 9.5.27)
  - Upgrading psy/psysh (v0.11.9 => v0.11.10)
  - Upgrading ramsey/collection (1.2.2 => 1.3.0)
  - Upgrading ramsey/uuid (4.6.0 => 4.7.0)
  - Upgrading roave/security-advisories (dev-latest 0399700 => dev-latest e8f9547)
  - Upgrading slevomat/coding-standard (8.6.4 => 8.7.1)
  - Upgrading symfony/beanstalkd-messenger (v6.1.3 => v6.2.0)
  - Upgrading symfony/browser-kit (v6.1.3 => v6.2.0)
  - Upgrading symfony/cache (v6.1.7 => v6.2.4)
  - Upgrading symfony/cache-contracts (v3.1.1 => v3.2.0)
  - Upgrading symfony/console (v6.1.7 => v6.2.3)
  - Upgrading symfony/css-selector (v6.1.3 => v6.2.3)
  - Upgrading symfony/deprecation-contracts (v3.1.1 => v3.2.0)
  - Upgrading symfony/dom-crawler (v6.1.7 => v6.2.3)
  - Upgrading symfony/event-dispatcher (v6.1.0 => v6.2.2)
  - Upgrading symfony/event-dispatcher-contracts (v3.1.1 => v3.2.0)
  - Upgrading symfony/expression-language (v6.1.6 => v6.2.2)
  - Upgrading symfony/filesystem (v6.1.5 => v6.2.0)
  - Upgrading symfony/finder (v6.1.3 => v6.2.3)
  - Upgrading symfony/intl (v6.1.7 => v6.2.0)
  - Upgrading symfony/lock (v6.1.7 => v6.2.0)
  - Upgrading symfony/mailer (v6.1.7 => v6.2.2)
  - Upgrading symfony/messenger (v6.1.6 => v6.2.2)
  - Upgrading symfony/mime (v6.1.7 => v6.2.2)
  - Upgrading symfony/options-resolver (v6.1.0 => v6.2.0)
  - Upgrading symfony/process (v6.1.3 => v6.2.0)
  - Upgrading symfony/property-access (v6.1.7 => v6.2.3)
  - Upgrading symfony/property-info (v6.1.7 => v6.2.3)
  - Upgrading symfony/rate-limiter (v6.1.3 => v6.2.2)
  - Upgrading symfony/redis-messenger (v6.1.7 => v6.2.0)
  - Upgrading symfony/serializer (v6.1.6 => v6.2.3)
  - Upgrading symfony/service-contracts (v3.1.1 => v3.2.0)
  - Upgrading symfony/stopwatch (v6.1.5 => v6.2.0)
  - Upgrading symfony/string (v6.1.7 => v6.2.2)
  - Upgrading symfony/translation (v6.1.6 => v6.2.3)
  - Upgrading symfony/translation-contracts (v3.1.1 => v3.2.0)
  - Upgrading symfony/validator (v6.1.7 => v6.2.3)
  - Upgrading symfony/var-dumper (v6.1.6 => v6.2.3)
  - Upgrading symfony/var-exporter (v6.1.3 => v6.2.3)
  - Upgrading symfony/yaml (v6.1.6 => v6.2.2)
  - Upgrading voku/portable-utf8 (6.0.10 => 6.0.11)
  - Upgrading wikimedia/composer-merge-plugin (dev-master 7692a62 => dev-master a4c4c62)
  - Upgrading zircote/swagger-php (4.5.1 => 4.5.3)
Writing lock file
Installing dependencies from lock file
Package operations: 0 installs, 15 updates, 0 removals
  - Downloading symfony/console (v6.2.3)
  - Downloading doctrine/lexer (2.1.0)
  - Downloading doctrine/instantiator (1.5.0)
  - Downloading doctrine/collections (2.1.2)
  - Downloading aws/aws-sdk-php (3.255.7)
  - Downloading symfony/translation (v6.2.3)
  - Downloading symfony/var-exporter (v6.2.3)
  - Downloading symfony/cache (v6.2.4)
  - Downloading ramsey/collection (1.3.0)
  - Downloading egulias/email-validator (3.2.4)
  - Downloading symfony/property-info (v6.2.3)
  - Downloading symfony/property-access (v6.2.3)
  - Downloading symfony/serializer (v6.2.3)
  - Downloading symfony/validator (v6.2.3)
  - Downloading symfony/finder (v6.2.3)
  - Upgrading symfony/console (v6.2.2 => v6.2.3): Extracting archive
  - Upgrading doctrine/lexer (1.2.3 => 2.1.0): Extracting archive
  - Upgrading doctrine/instantiator (1.4.1 => 1.5.0): Extracting archive
  - Upgrading doctrine/collections (2.1.1 => 2.1.2): Extracting archive
  - Upgrading aws/aws-sdk-php (3.255.3 => 3.255.7): Extracting archive
  - Upgrading symfony/translation (v6.2.2 => v6.2.3): Extracting archive
  - Upgrading symfony/var-exporter (v6.2.2 => v6.2.3): Extracting archive
  - Upgrading symfony/cache (v6.2.2 => v6.2.4): Extracting archive
  - Upgrading ramsey/collection (1.2.2 => 1.3.0): Extracting archive
  - Upgrading egulias/email-validator (3.2.1 => 3.2.4): Extracting archive
  - Upgrading symfony/property-info (v6.2.0 => v6.2.3): Extracting archive
  - Upgrading symfony/property-access (v6.2.0 => v6.2.3): Extracting archive
  - Upgrading symfony/serializer (v6.2.2 => v6.2.3): Extracting archive
  - Upgrading symfony/validator (v6.2.2 => v6.2.3): Extracting archive
  - Upgrading symfony/finder (v6.2.0 => v6.2.3): Extracting archive
Generating optimized autoload files
82 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
No security vulnerability advisories found

AzuraCast Setup
===============

Welcome to AzuraCast. Please wait while some key dependencies of AzuraCast are set up...

Initialize AzuraCast
====================

Initializing essential settings...
 * Environment: Production
 * Installation Method: Docker


Running Database Migrations
---------------------------


Database Migrations
===================

[2022-12-31T01:10:04.994203+00:00] AzuraCast.ERROR: Doctrine\DBAL\Exception\ConnectionException: An exception occurred in the driver: SQLSTATE[HY000] [1045] Access denied for user 'azuracast'@'localhost' (using password: YES) (uncaught exception) at /var/azuracast/www/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php line 101 while running console command `azuracast:setup` {"exception":"[object] (Doctrine\\DBAL\\Exception\\ConnectionException(code: 1045): An exception occurred in the driver: SQLSTATE[HY000] [1045] Access denied for user 'azuracast'@'localhost' (using password: YES) at /var/azuracast/www/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php:101)\n[previous exception] [object] (Doctrine\\DBAL\\Driver\\PDO\\Exception(code: 1045): SQLSTATE[HY000] [1045] Access denied for user 'azuracast'@'localhost' (using password: YES) at /var/azuracast/www/vendor/doctrine/dbal/src/Driver/PDO/Exception.php:28)\n[previous exception] [object] (PDOException(code: 1045): SQLSTATE[HY000] [1045] Access denied for user 'azuracast'@'localhost' (using password: YES) at /var/azuracast/www/vendor/doctrine/dbal/src/Driver/PDO/MySQL/Driver.php:28)"} []
[2022-12-31T01:10:04.994897+00:00] AzuraCast.WARNING: Console command `azuracast:setup` exited with error code 1045. [] []

In ExceptionConverter.php line 101:

  An exception occurred in the driver: SQLSTATE[HY000] [1045] Access denied for user 'azuracast'@'localhost' (using pas
  sword: YES)


In Exception.php line 28:

  SQLSTATE[HY000] [1045] Access denied for user 'azuracast'@'localhost' (using password: YES)


In Driver.php line 28:

  SQLSTATE[HY000] [1045] Access denied for user 'azuracast'@'localhost' (using password: YES)


azuracast:setup [--update] [--load-fixtures] [--release]

ERROR: 255
Creating azuracast ... done
```

**Comment by fnfzlznddzn** [comment]
Just tried to install a new azuracast and got the same errors.

**Comment by ItsMitchh** [comment]
We've had this issue crop up a few times, its a rare one and a even harder to reproduce, I'll raise it up anyway.

**Comment by Hi-ImKyle** [comment]
I figured with it being SQL problems I tried [this](https://github.com/AzuraCast/AzuraCast/issues/5769#issuecomment-1272468980) but that didn't do anything.

**Comment by fafamonge** [comment]
I have the same problem here https://github.com/AzuraCast/AzuraCast/issues/5916

**Comment by fnfzlznddzn** [comment]
I looked this issue up and found quite a few threads (not related to azuracast specifically), for example: https://stackoverflow.com/questions/59838692/mysql-root-password-is-set-but-getting-access-denied-for-user-rootlocalhost

Could it be related to the mysql default passwords being too complex (special characters, ...) ? Some users in this thread report that this was their issue.

**Comment by Hi-ImKyle** [comment]
```
docker-compose down -v 
docker-compose up -d
```

Well I'll be damned. This works for me. If it is the case of the mysql passwords being too complex, then it's most likely never the users fault, at least was with my case, I was never setting the password. 

I will leave this open till the devs can find the source of the problem, or if you want to close it then do so.

**Comment by BusterNeece** [comment]
This is very strange, because we store the database credentials in the environment file _outside_ the Docker ecosystem, so flushing the volumes as the commands above suggest should have absolutely no impact on that. It would just force the database to recreate with the password that's in the environment file, thus "fixing" the problem in a roundabout way by making the new database have the correct password.

I think there may be an issue in some cases where the installer script is, for some reason, losing the existing environment file and thinking a password hasn't been generated yet, then generating a new one.

**Comment by BusterNeece** [comment]
While I don't know exactly what's causing this still, I've implemented some changes to our installer so that if it modifies the environment files, it'll automatically save backup copies of the originals before overwriting them. If this happens, this could mean that your MariaDB root password was, for whatever reason, reset by the system to a new random value, which is causing this problem. Look through the other files in your `/var/azuracast` directory, and you should see the backup file, which you can restore to the main filename, then use `docker-compose down` (note, NOT `-v`) and `docker-compose up -d` to restart the services.

**Comment by robz0** [comment]
Best on a clean install...

My Misconception:
I had this problem trying Docker install with AzuraCast provided docker.sh script, on an Oracle instance.Being new to cloud computing - as a CPanel/shared hosting hobbyist - I installed HestiaCP on the instance when I was faced with an empty 'box'.  I wanting to manage it and it it up but did not realise that we do not need any of that! I think the first install of AzuraCast may have worked but reboots/upgrades etc. it all fell apart.

Solution for me: 
It turns out that AzuraCast IS THE BOX - as promised.

So I started again. Just following the Install instructions on the new, clean instance (no Control panels or anything other than OS installed) it worked. Managed to reboot and go through the docker.sh commands without error. 

Answer (as said in other comments and threads on this error) 
- Install on a clean install to avoid conflicts - probably especially no Control Panels. AzuraCast sets everything up. Thanks @BusterNeece and team.
